CoreScriptsRoot/Modules/Settings/Pages/Players.lua

```diff
@@ -117,12 +134,42 @@ local function Initialize()
 					if friendLabel and friendLabelText and friendLabelText.Text ~= '' then
 						friendLabel.ImageTransparency = 1
 						friendLabelText.Text = ''
-						if localPlayer and player then
+					end
+					if AtFriendLimit(localPlayer) then
+						while PromptCreator:IsCurrentlyPrompting() do
+							wait()
+						end
+						PromptCreator:CreatePrompt({
```

Comment:
Nit: There is a good bit of repetition between both these PromptCreator calls, could you just check if either player is at the friend limit at the beginning of the addFriendFunc and customize the text only based on which player is over the limit.